example/fft_benchmark.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# Copyright NumFOCUS
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script for benchmarking accelerated ITK FFT performance
+
+import time
+
+import numpy as np
+import itk
+itk.auto_progress(2)
+
+pixel_type = itk.F
+dimension = 3
+image_type = itk.Image[pixel_type, dimension]
+complex_image_type = itk.Image[itk.complex[pixel_type],dimension]
+fft_filter_type = itk.ForwardFFTImageFilter[image_type,complex_image_type]
+
+# Return time for filter.Update()
+def benchmark_fft(itk_fft_filter:fft_filter_type) -> float:
+    start = time.time()
+    itk_fft_filter.Update()
+    end = time.time()
+    return end - start
+
+def run_fft(itk_fft_filter_type, image_size:list()) -> float:
+    fft_filter = itk_fft_filter_type.New()
+    image = itk.image_from_array(np.random.rand(image_size[0],image_size[1],image_size[2]).astype(np.float32))
+    fft_filter.SetInput(image)
+    return benchmark_fft(fft_filter)
+
+# One side of 3D image
+# Total volume of cube is image_len ^ 3 voxels
+image_lens = [10, 30, 100, 200, 300, 600, 800, 1000, 1200, 1500, 2000]
+
+# ITK default CPU implementation
+vnl_type = itk.VnlForwardFFTImageFilter[image_type, complex_image_type]
+# VkFFT accelerated implementation
+vk_type = itk.VkForwardFFTImageFilter[image_type]
+
+itk.auto_progress(0)
+
+print(f'Experiment   Image Len (px)\tVolume (px)\tCPU FFT Time (s)\tGPU FFT Time(s)\tSpeedup')
+for idx, image_len in enumerate(image_lens):
+    image_size = [image_len] * 3
+    n_iterations = 3    
+
+    for iteration in range(n_iterations):
+         vnl_interval = run_fft(vnl_type, image_size)
+         vk_interval = run_fft(vk_type, image_size)
+         speedup = 1 - (vk_interval / vnl_interval)
```

Comment:
This is fine as is, especially if you are mimicking other code in ITK.  Nonetheless, in many contexts, the performance gain is instead defined as `vnl_interval / vk_interval - 1`.  So that a 100% improvement means that it runs in half the time.

Reply:
This formula was off-the-cuff, I can update to match literature. I will say that the current implementation seems more intuitive to me as "100%" speedup is a bounds approaching 0 sec runtime, whereas in the proposed implementation the speedup approaches infinity%. "1000% speedup over the current implementation" seems less intuitive than "90% speedup".

Sample output with new formula:
```
Experiment   Image Len (px)     Volume (px)     CPU FFT Time (s)        GPU FFT Time(s) Speedup
         0                 10   1.00e+03                0.032390        0.746716        -95.7%
         0                 10   1.00e+03                0.000164        0.171591        -99.9%
         0                 10   1.00e+03                0.000117        0.138355        -99.9%
         1                 30   2.70e+04                0.001242        0.454585        -99.7%
         1                 30   2.70e+04                0.002063        0.161120        -98.7%
         1                 30   2.70e+04                0.001266        0.136933        -99.1%
         2                100   1.00e+06                0.042140        0.492682        -91.4%
         2                100   1.00e+06                0.062667        0.156256        -59.9%
         2                100   1.00e+06                0.041899        0.147126        -71.5%
         3                200   8.00e+06                0.433797        0.442058        -1.9%
         3                200   8.00e+06                0.449701        0.200594        124.2%
         3                200   8.00e+06                0.364515        0.172711        111.1%
         4                300   2.70e+07                1.381301        1.163240        18.7%
         4                300   2.70e+07                1.310901        0.296672        341.9%
         4                300   2.70e+07                1.224416        0.296312        313.2%
         5                600   2.16e+08                11.413113       2.275016        401.7%
         5                600   2.16e+08                11.398046       1.478473        670.9%
         5                600   2.16e+08                11.405163       1.458265        682.1%
         6                800   5.12e+08                44.632636       3.741074        1093.0%
```

Reply:
1093.0% speedup = 11x faster, meaning it takes 11 times less time to compute = 11x shorter.

Reply:
Although, slowdowns are usually expressed by small positive numbers. This is accomplished by having relative speed formula = vnl_interval / vk_interval. In this case 100% means equal speed.

Reply:
Makes sense. As long as this is more readable for others I am happy with the change then. 😃